CONTRIBUTING.md

```diff
@@ -101,9 +101,87 @@ the Apache license found in the LICENSE file.
 A number of issues in the 
 [issue tracker](https://github.com/GoogleCloudPlatform/gcloud-intellij/issues/new)
 have been tagged as "[Help Wanted](https://github.com/GoogleCloudPlatform/gcloud-intellij/labels/help%20wanted)." 
-These are relatively small, self-contained changes that are good places to start. 
+These are relatively small, self-contained changes that are good places to start.
+
+## Project Utilities
+
+### `CloudToolsRule`
+We have a custom JUnit `TestRule` that reduces the amount of boilerplate code we have in our unit
+tests. Notably, this rule handles a lot of the set-up and tear-down required for a proper IntelliJ
+project context.
+
+For example:
+
+```
+@Rule public final CloudToolsRule cloudToolsRule = new CloudToolsRule(this);
+```
+
+A summary of its features can be found below.
+
+**Always:**
+1. Initializes the fields annotated with `@Mock` by calling
+   `MockitoAnnotations.initMocks(testInstance)`
+2. Creates an `IdeaProjectTestFixture`, which sets up static application state for the IntelliJ SDK
+   and creates a `Project`
+
+**Additional Annotations:**
+
+`@MockService`:
+- Annotates fields of any type, but should also have the `@Mock` annotation
```

Comment:
Yeah you can. We can prevent it and throw an exception if the `@Mock` annotation is missing, but it's not _that_ important because it technically doesn't matter how it's mocked, or even if it isn't mocked and it's a real object. It just does a simple swap and doesn't care what the value really is.